hummingbot/market/kraken/kraken_constants.py

```diff
@@ -24,10 +24,16 @@
     "AUD"
 ]
 
-FIAT_QUOTES = FIAT_QUOTES + ["Z" + quote for quote in FIAT_QUOTES]
+FIAT_QUOTES = ["Z" + quote for quote in FIAT_QUOTES] + FIAT_QUOTES
```

Comment:
this change is good. however would it mean that `convert_from_exchange_trading_pair("XTZUSD") == "XT-ZUSD"` ?

Reply:
I see, that is one exception. But it is best to logically let the quotes with the `Z` prefix come first so majority of all pairs can work properly with `split_to_base_quote(exchange_trading_pair)`

Reply:
Yes definitely this change (`Z` prefix quotes first) is an improvement. We do need to have a more complete solution though.

Reply:
An ideal solution would allow us to verify that the base is actually an asset once we've taken away the possible quote. e.g. we test the split XTZUSD->XT+ZUSD, we check XT and find it isn't an asset, we continue, then we check XTZUSD->XTZ+USD, we check XTZ and find it is an asset. Perhaps caching the results of https://api.kraken.com/0/public/Assets including altnames would be one way?

Reply:
Exactly the main reason I added something like `if len(quote) > 2 or quote == "SC"` in `split_to_base_quote(exchange_trading_pair)`. Because `SC` is the only Asset with 2 letter.

Reply:
> 
> 
> Exactly the main reason I added something like `if len(quote) > 2 or quote == "SC"` in `split_to_base_quote(exchange_trading_pair)`. Because `SC` is the only Asset with 2 letter.

I believe just checking if the length of base is greater than 2 with the exception of "SC" asset is a better alternative than caching the result of  https://api.kraken.com/0/public/Assets and running a check on every if it exists on every available asset.